aderyn_core/src/visitor/ast_visitor.rs

```diff
@@ -497,15 +497,26 @@ pub trait ASTConstVisitor {
     ) -> Result<()> {
         Ok(())
     }
+
+    fn visit_node_id(&mut self, _node_id: Option<NodeID>) -> Result<()> {
+        Ok(())
+    }
 }
 
 pub trait Node {
+    /// [`Node::accept`] is designed to propagate
     fn accept(&self, _visitor: &mut impl ASTConstVisitor) -> Result<()> {
         Ok(())
     }
+    /// [`Node::accept_metadata`] is designed to propagate into the AST subtree
+    /// although it doesn't happen by itself. [`Node::accept`] triggers the propagation
     fn accept_metadata(&self, _visitor: &mut impl ASTConstVisitor) -> Result<()> {
         Ok(())
     }
+    /// [`Node::accept_id`] is not designed to propagate into the AST subtree
+    fn accept_id(&self, _visitor: &mut impl ASTConstVisitor) -> Result<()> {
+        Ok(())
+    }
 }
```

Comment:
@alexroan  Here I have documented the fact that `accept_id()` is not propagated (so there is no front processing)

Reply:
Speed tradeoff in terms of compilation time may increase tho : ) Cuz the compiler has to create these methods  for all types that implement Node :P

Reply:
You mean compilation of `aderyn` tho, right? That's completely and utterly fine :D